core/market/decentralized/src/negotiation/error.rs

```diff
@@ -98,42 +93,34 @@ pub enum QueryEventsError {
 
 #[derive(Error, Debug)]
 pub enum ProposalError {
-    #[error("Subscription [{0}] was already unsubscribed.")]
-    Unsubscribed(SubscriptionId),
+    #[error("Proposal subscription Offer error: [{0}].")]
+    QueryOfferError(#[from] QueryOfferError),
+    #[error("Proposal subscription Demand error: [{0}].")]
+    DemandError(#[from] DemandError),
     #[error("Subscription [{0}] expired.")]
     SubscriptionExpired(SubscriptionId),
-    #[error("Proposal [{0}] not found for subscription [{1}].")]
-    ProposalNotFound(ProposalId, SubscriptionId),
-    #[error("Failed to get Proposal [{0}] for subscription [{1}]. Error: [{2}]")]
-    FailedGetProposal(ProposalId, SubscriptionId, DbError),
+    #[error("Proposal [{0}] not found for subscription [{1:?}].")]
+    NotFound(ProposalId, Option<SubscriptionId>),
+    #[error("Failed to get Proposal [{0}] for subscription [{1:?}]. Error: [{2}]")]
+    Get(ProposalId, Option<SubscriptionId>, DbError),
     #[error("Failed to save counter Proposal for Proposal [{0}]. Error: {1}")]
-    FailedSaveProposal(ProposalId, DbError),
+    Save(ProposalId, DbError),
     #[error("Failed to send counter Proposal for Proposal [{0}]. Error: {1}")]
-    FailedSendProposal(ProposalId, ProtocolProposalError),
+    Send(ProposalId, ProtocolProposalError),
+    #[error("Internal error: {0}.")]
+    InternalError(#[from] ErrorMessage),
```

Comment:
Proposition for future: introduce generic (not bound to `Proposal`) `InternalError` with options: `Get`, `Save`, `Unexpected` and so on